src/main/java/net/sf/jabref/exporter/SaveDatabaseAction.java

```diff
@@ -167,6 +169,66 @@ public void run() {
 
             if (FileBasedLock.waitForFileLock(panel.getBibDatabaseContext().getDatabaseFile(), 10)) {
+                // Check for external modifications to alleviate multiuser concurrency issue when near
+                // simultaneous saves occur to a shared database file:
+                if (panel.isUpdatedExternally()
+                        || Globals.fileUpdateMonitor.hasBeenModified(panel.getFileMonitorHandle())) {
+                    String[] opts = new String[] {Localization.lang("Review changes"), Localization.lang("Save"),
+                            Localization.lang("Cancel")};
+                    int answer = JOptionPane.showOptionDialog(panel.frame(),
+                            Localization.lang("File has been updated externally. " + "What do you want to do?"),
+                            Localization.lang("File updated externally"), JOptionPane.YES_NO_CANCEL_OPTION,
+                            JOptionPane.QUESTION_MESSAGE, null, opts, opts[0]);
+
+                    if (answer == JOptionPane.CANCEL_OPTION) {
+                        cancelled = true;
+                        return;
+                    } else if (answer == JOptionPane.YES_OPTION) {
+                        cancelled = true;
+
+                        JabRefExecutorService.INSTANCE.execute((Runnable) () -> {
+
+                            if (!FileBasedLock.waitForFileLock(panel.getBibDatabaseContext().getDatabaseFile(), 10)) {
+                                // TODO: GUI handling of the situation when the externally modified file keeps being locked.
+                                LOGGER.error("File locked, this will be trouble.");
+                            }
+
+                            ChangeScanner scanner = new ChangeScanner(panel.frame(), panel,
+                                    panel.getBibDatabaseContext().getDatabaseFile());
+                            JabRefExecutorService.INSTANCE.executeWithLowPriorityInOwnThreadAndWait(scanner);
+                            if (scanner.changesFound()) {
+                                scanner.displayResult((ChangeScanner.DisplayResultCallback) resolved -> {
+                                    if (resolved) {
+                                        panel.setUpdatedExternally(false);
+                                        SwingUtilities.invokeLater(
+                                                (Runnable) () -> panel.getSidePaneManager().hide("fileUpdate"));
+                                    } else {
+                                        cancelled = true;
+                                    }
+                                });
+                            }
+                        });
+
+                        return;
+                    } else { // User indicated to store anyway.
+                                 // See if the database has the protected flag set:
+                        List<String> pd = panel.getBibDatabaseContext().getMetaData()
```

Comment:
There is a method `isProtected` (or something like that) in `MetaData` which is (now) the preferred way to receive this information.